include/teqp/models/multifluid_ecs_mutant.hpp

```diff
@@ -0,0 +1,178 @@
+            The mixture model depends on coefficients for a temperature and density polynomial. \n
+            Only valid for binary mixtures. \n
+            The pECS reducing function follow the mathematical operations: \n
+            \f$F(\bar x, Y, T, \rho ) = \sum_{i=1}^{N} x_{i}^{2} Y_{\mathrm{c},i} + \sum_{i=1}^{N-1}\sum_{j=i+1}^{N} 2 x_{i} x_{j}  Y_{ij} f_{Y}(T,\rho).\f$ \n
+            \f$f_{T,v} = \sum_{i}^{n} \sum_{j}^{m-i} c_{ij,T,v} \tau_{\mathrm{ECS}}^{i} \delta_{\mathrm{ECS}}^{j}\f$ \n
+            \f$\tau_{\mathrm{ECS}} = \frac{Y_{T,ij}}{T}\f$ \n
+            \f$\delta_{\mathrm{ECS}} = \rho Y_{v,ij}\f$ \n
+            \f$Y_{T,ij} =  \sqrt{T_{\mathrm{c},i} T_{\mathrm{c},j}}\f$ \n
+            \f$Y_{v,ij} =  \left( \frac{1}{\rho_{\mathrm{c},i}^{1/3}} + \frac{1}{\rho_{\mathrm{c},j}^{1/3}}\right)^{3}\f$ \n
+            \f$c_{ij,T,v} = a_{ij,1} + a_{ij,2} x_{i} + a_{ij,3} x_{i}^{2}\f$. 
+    */
+    class Reducing_ECS {
+
+    private:
+        /*!
+          Matrix containing the coefficients for the temperature polynominal: \f$a_{ij,T}\f$
+        */
+        Eigen::MatrixXd  tr_coeffs;
+        /*!
+                  Matrix containing the coefficients for the volume polynominal: \f$a_{ij,v}\f$
+        */
+        Eigen::MatrixXd dr_coeffs;
+
+    public:
+        /*!
+          Critical temperature array of corresponding pure fluid contributions.
+        */
+        Eigen::ArrayXd Tc;
+        /*!
+          Critical volume array of corresponding pure fluid contributions.
+        */
+        Eigen::ArrayXd vc;
+        template<typename ArrayLike>
+        Reducing_ECS(const ArrayLike& Tc, const ArrayLike& vc, const nlohmann::json& jj) : Tc(Tc), vc(vc) {
+
+            if (not jj.contains("tr_coeffs")) {
+                throw teqp::InvalidArgument("tr_coeffs not in provided json");
+            }
+
+            if (not jj.contains("dr_coeffs")) {
+                throw teqp::InvalidArgument("dr_coeffs not in provided json");
+            }
+
+            auto json_tr_coeffs = jj.at("tr_coeffs");
+            auto json_dr_coeffs = jj.at("dr_coeffs");
+
+            int rows_tr = json_tr_coeffs.size();
+            int cols_tr = json_tr_coeffs[0].size();
+            tr_coeffs.resize(rows_tr, cols_tr);
+
+            int rows_dr = json_dr_coeffs.size();
+            int cols_dr = json_dr_coeffs[0].size();
+            dr_coeffs.resize(rows_dr, cols_dr);
+
+
+            for (auto i = 0; i < rows_tr; ++i) {
+                for (auto j = 0; j < cols_tr; ++j) {
+                    tr_coeffs(i, j) = json_tr_coeffs[i][j];
+                }
+            }
+
+            for (auto i = 0; i < rows_dr; ++i) {
+                for (auto j = 0; j < cols_dr; ++j) {
+                    dr_coeffs(i, j) = json_dr_coeffs[i][j];
+                }
+            }
+
+        }
+
+        /*!
+            Reducing function for temperature
+        */
+        template <typename TTYPE, typename RHOTYPE, typename MoleFractions>
```

Comment:
You could (like I did in the multifluid case) write a generic Y_r function and use it in both reducing functions since they are identical. I think that is worthwhile

Reply:
Or maybe not, since the cross term is different. I'll leave that up to you.